src/Input/InputParamTrait.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+/**
+ * @see       https://github.com/laminas/laminas-cli for the canonical source repository
+ * @copyright https://github.com/laminas/laminas-cli/blob/master/COPYRIGHT.md
+ * @license   https://github.com/laminas/laminas-cli/blob/master/LICENSE.md New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Laminas\Cli\Input;
+
+use ArrayObject;
+use InvalidArgumentException;
+use Symfony\Component\Console\Helper\QuestionHelper;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+use function sprintf;
+
+trait InputParamTrait
+{
+    /**
+     * @internal
+     * @var null|ArrayObject<string, InputParam>
+     */
+    private $inputParams;
+
+    /**
+     * @param null|mixed $default
+     * @return $this
+     */
+    final public function addParam(
+        string $name,
+        string $description,
+        string $type,
+        bool $required = false,
+        $default = null,
+        array $options = []
+    ) : self {
+        $mode = $type === InputParam::TYPE_BOOL
+            ? InputOption::VALUE_NONE
+            : InputOption::VALUE_REQUIRED;
+
+        $this->addOption(
```

Comment:
We need to add an abstract member for this, as we cannot guarantee it is available in the parent.

```php
abstract public function addOption(string $name, $shortcut = null, int $mode = null, string $description = '', $default = null);
```

Reply:
... and we do not need to wait for the next major, we already have the issue because the signature has changed between v4 and v5 (types added).

I believe we still should support v4, so I think we have to remove the abstract from trait... Thoughts?

Reply:
Okay. But we should add a `@todo` to the trait indicating we will need to add them once we update to a symfony/console minimum version of v5 or later.